### PR TITLE
- adding v4 .Net lambda sample project

### PR DESCRIPTION
**Pull request by sbhuller**

### CODE REVIEW

File: RightslineSampleLambdaDotNetV4/RightslineSampleLambdaDotNetV4/Function.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+                        ""start"":0,""rows"":25,""sortOrders"":[""sequencenumber asc""]}";
+
+                    var catalogResults = await this._v4.Search("catalog-item", catalogPayload);
+
+                    if (catalogResults.Count() == 0)
+                    {
+                        Console.WriteLine($"Cannot find Catalog Items with template id {messageEntity.Entity.Template.TemplateId}.");
+                        Console.WriteLine("Lambda processing aborted");
+                    }
+
```
Comment (ckissinger) [inline RightslineSampleLambdaDotNetV4/RightslineSampleLambdaDotNetV4/Function.cs:106]: maybe add an example of grabbing additional information from a rightset in here also? or getting all catalog items associated to a rightset when it is changed?

File: RightslineSampleLambdaDotNetV4/RightslineSampleLambdaDotNetV4/Models/EntityRelationshipModel.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
+﻿using RightslineSampleLambdaDotNetV4.Consts;
+using System;
+
+namespace RightslineSampleLambdaDotNetV4.Models
+{
+    public class EntityRelationshipModel
+    {
+        public int? Id { get; set; }
+        public string ParentURL { get; set; }
```
Comment (ckissinger) [inline RightslineSampleLambdaDotNetV4/RightslineSampleLambdaDotNetV4/Models/EntityRelationshipModel.cs:9]: I think this is the v3 model, in v4 we moved away from these URL string properties and we also added the template and status of the parent and child

File: RightslineSampleLambdaDotNetV4/RightslineSampleLambdaDotNetV4/Enumerations/Enumerations.cs [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+using System.Reflection;
+
+namespace RightslineSampleLambdaDotNetV4
+{
+    public enum CharTypeID
+    {
+        [Description("catalog-item")]
+        CatalogItem = 1,
+        [Description("rightset")]
+        Rightset = 3,
```
Comment (adammorton9) [inline RightslineSampleLambdaDotNetV4/RightslineSampleLambdaDotNetV4/Enumerations/Enumerations.cs:12]: Use "right" instead of "rightset"

Comment (sbhuller) [reply RightslineSampleLambdaDotNetV4/RightslineSampleLambdaDotNetV4/Enumerations/Enumerations.cs]: Resolved in branch.

**Review by adammorton9 — APPROVED** [review]

**Pull request merged**